server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -256,44 +256,52 @@ public boolean equalsAliases(MetaData other) {
      * @return a map of index to a list of alias metadata, the list corresponding to a concrete index will be empty if no aliases are
      * present for that index
      */
-    public ImmutableOpenMap<String, List<AliasMetaData>> findAliases(final String[] aliases, String[] concreteIndices) {
+    public ImmutableOpenMap<String, List<AliasMetaData>> findAliases(String[] aliases, String[] concreteIndices) {
+        return findAliases(false, aliases, concreteIndices);
+    }
+
+    /**
+     * Finds the specific index aliases that match with the specified aliases directly or partially via wildcards and
+     * that point to the specified concrete indices or match partially with the indices via wildcards.
+     *
+     * @param aliasesProvided Whether concrete aliases where provided by the user in the original request. This information
+     *                        gets lost as internally aliases may get replaced.
+     * @param aliases         The names of the index aliases to find
+     * @param concreteIndices The concrete indexes the index aliases must point to order to be returned.
+     * @return a map of index to a list of alias metadata, the list corresponding to a concrete index will be empty if no aliases are
+     * present for that index
+     */
+    public ImmutableOpenMap<String, List<AliasMetaData>> findAliases(boolean aliasesProvided, String[] aliases, String[] concreteIndices) {
         assert aliases != null;
         assert concreteIndices != null;
         if (concreteIndices.length == 0) {
             return ImmutableOpenMap.of();
         }
 
-        boolean matchAllAliases = matchAllAliases(aliases);
         ImmutableOpenMap.Builder<String, List<AliasMetaData>> mapBuilder = ImmutableOpenMap.builder();
         Iterable<String> intersection = HppcMaps.intersection(ObjectHashSet.from(concreteIndices), indices.keys());
         for (String index : intersection) {
             IndexMetaData indexMetaData = indices.get(index);
             List<AliasMetaData> filteredValues = new ArrayList<>();
             for (ObjectCursor<AliasMetaData> cursor : indexMetaData.getAliases().values()) {
                 AliasMetaData value = cursor.value;
-                if (matchAllAliases || Regex.simpleMatch(aliases, value.alias())) {
+                if (Strings.isAllOrWildcard(aliases) || Regex.simpleMatch(aliases, value.alias())) {
```

Comment:
this is not exactly the same as the previous matchAllAliases method. I always wondered why, so maybe a good time to fix it, but this will change the behaviour in subtle ways when a '*' is provided or when an array the contains '_all' with more than one item is provided..

Reply:
> but this will change the behaviour in subtle ways when a '*' is provided or when an array the contains '_all' with more than one item is provided

If _all, * with other alias names is provided then `Strings.isAllOrWildcard(...)` return false, so I think that is ok? I will also look into if it is possible to change TransportGetAliasesAction so that we don't have to change this method, like I mentioned in the other comment.

Reply:
but the previous matchAllAliases didn't look for the wildcard, and only looked at whether _all was anywhere within the array.

Reply:
> and only looked at whether _all was anywhere within the array

Or whether the array was an empty array, which sort of matched with the logic inside `RestGetAliasesAction`.

If you take a look at `RestGetAliasesAction` at line 81 where `paramAsStringArrayOrEmptyIfAll(...)` gets execute, it will replace `_all` and `*` with an empty array.

Reply:
good point. ok then we can say that we are making transport client work the same way as REST, that should be good